### ISSUE TITLE
Refactor backend

### ISSUE THREAD

**Issue by adamf321** [wearenolte/lean-theme#165]
The backend functions and classes of Lean have gotten a bit messy over time. We should go through and simplify all of this.

**Comment by Fcog** [comment]
yes, I think we should unify all backend functionality in a single composer library

**Comment by Fcog** [comment]
maybe this library can help https://framework.themosis.com/
